### PR TITLE
Remove crm_perror from tools & daemons

### PR DESCRIPTION
**Pull request by clumens**

### CODE REVIEW

File: daemons/based/based_io.c [hunk 1]
```diff
@@ -50,7 +50,7 @@ cib_rename(const char *old)
     new_fd = mkstemp(new);
     crm_err("Archiving unusable file %s as %s", old, new);
     if ((new_fd < 0) || (rename(old, new) < 0)) {
-        crm_perror(LOG_ERR, "Couldn't rename %s as %s", old, new);
+        crm_err("Couldn't rename %s as %s: %s", old, new, pcmk_rc_str(errno));
```
Comment (kgaillot) [inline daemons/based/based_io.c:53]: This has an additional problem -- the first crm_err() could theoretically change errno.

We could save errno right after mkstemp(), but I think it doesn't make sense to log an "Archiving" error then a "Couldn't rename" error -- let's just do `if ... "Couldn't archive unusable file %s (disabling disk writes and continuing)" ... else ... "Archived unusable file ..."`

File: daemons/execd/remoted_tls.c [hunk 2]
```diff
@@ -273,39 +273,39 @@ bind_and_listen(struct addrinfo *addr)
 
     fd = socket(addr->ai_family, addr->ai_socktype, addr->ai_protocol);
     if (fd < 0) {
-        crm_perror(LOG_ERR, "Listener socket creation failed");
-        return -1;
+        crm_err("Listener socket creation failed: %", pcmk_rc_str(errno));
+        return -errno;
```
Comment (kgaillot) [inline daemons/execd/remoted_tls.c:277]: All these returns will have the same problem ... need to save errno immediately after the system call

**Review by kgaillot — COMMENTED** [review]
Nice, a lot of this was broken to begin with ...

File: daemons/execd/remoted_tls.c (hunk 2 above)
Comment (clumens) [reply daemons/execd/remoted_tls.c:277]: Fixed.

File: daemons/based/based_io.c (hunk 1 above)
Comment (clumens) [reply daemons/based/based_io.c:53]: I thought that one was pretty weird, too.  Fixed.

**Review by kgaillot — APPROVED** [review]

**Pull request merged**